gammapy/obs/datastore.py

```diff
@@ -0,0 +1,189 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import os
+from astropy.table import Table
+from astropy.coordinates import SkyCoord, Angle
+from ..catalog import select_sky_box, skycoord_from_table
+
+__all__ = ['DataStore', 'DataStoreIndexTable']
+
+
+def _make_filename_hess_scheme(obs_id, filetype='events'):
+    """Make filename string for the HESS storage scheme.
+
+    Parameters
+    ----------
+    obs_id : int
+        Observation ID
+    filetype : {'events', 'effective area', 'psf', 'background'}
+        Type of file
+
+    Examples
+    --------
+    >>> _make_filename_hess_scheme(obs_id=89565, filetype='effective area')
+    'run089400-089599/run089565/hess_aeff_2d_089565.fits.gz'
+    """
+    obs_id_min = obs_id - (obs_id % 200)
+    obs_id_max = obs_id_min + 199
+    group_folder = 'run{:06d}-{:06d}'.format(obs_id_min, obs_id_max)
+    obs_folder = 'run{:06d}'.format(obs_id)
+
+    if filetype == 'events':
+        label = 'events'
+    elif filetype == 'psf':
+        label = 'psf_king'
+    elif filetype == 'effective area':
+        label = 'aeff_2d'
+    elif filetype == 'background':
+        label = 'bkg_offruns'
+    else:
+        raise ValueError('Unknown filetype: {}'.format(filetype))
+
+    filename = 'hess_{}_{:06d}.fits.gz'.format(label, obs_id)
+
+    return os.path.join(group_folder, obs_folder, filename)
+
+
+class DataStoreIndexTable(Table):
```

Comment:
@taldcroft - This is the Astropy Table subclass I mentioned [here](https://groups.google.com/d/msg/astropy-dev/0EaOw9peWSk/6Jj26gcC_O8J) where I want to do some cleanup on read.

I'm still not sure what the best approach is:
1. Overwrite  `__init__` and call `_init_cleanup` from there?
2. Overwrite `read` and call `_init_cleanup` from there? (this is what I do at the moment and it seems to work OK, but it does make an extra copy just so that I can change the type to be `DataStoreIndexTable` and not `Table`)
3. Add a `read_and_cleanup` method, i.e. the user has to specify that they want to read a FITS table that needs cleanup.

The use case here is that I have the same table info in somewhat different formats, and I want to normalize it to the format I like on read.